### ISSUE TITLE
False positive "unresolved macro" for syn::custom_punctuation

### ISSUE THREAD

**Issue by nathanwhit** [rust-lang/rust-analyzer#10128]
Usage of the[ `custom_punctuation` macro](https://docs.rs/syn/1.0.75/syn/macro.custom_punctuation.html) from `syn` results in two false positive unresolved macro diagnostics:
``unresolved macro  `spans!`  ``  and ``unresolved macro `__S!` ``

The strange part is that running the new `Expand macros recursively` command works fine, and the output doesn't contain any uses of either of the "unresolved" macros. This reproduces on both the latest nightly (9/1/2021) and the latest stable version of RA.

To reproduce:
- Add `syn` as a dependency
- Invoke `syn::custom_punctuation`, e.g.
```rust
syn::custom_punctuation!(LeftRightArrow, <=>);
```
